task2.js

```diff
@@ -0,0 +1,23 @@
+import {EventEmitter} from "./task1.js";
+
+class WithTime extends EventEmitter {
+  execute(asyncFunc, ...args) {
+    this.emit("begin");
+    asyncFunc(...args).then((res) => {
+      this.emit("end");
+      return res.json();
+    }).then((data) => {
+      this.emit("data", data);
+    });
+  }
```

Comment:
Per task `execute` function should **compute the time taken by this function**. Maybe I look the wrong way, but I can't see computing time taken by this function in your code.